File: typescript/src/main/vsdk/toolkit/raycasting/photonMap/Importon.ts
import { Vector3D } from "../../common/linealAlgebra/Vector3D";
import { IrrPhoton } from "./IrrPhoton";

// Importon: identical to IrrPhoton, but with some extra functions
export class Importon extends IrrPhoton {
  public SetAll(imp: number, pot: number, foot: number): void {
    void pot;
    void foot;
    this.m_power.r = imp;
  }

  public PSetAll(imp: number, pot: number, foot: number): void {
    void pot;
    void foot;
    this.m_irradiance.r = imp;
  }

  public constructor();
  public constructor(
    pos: Vector3D,
    importance: number,
    potential: number,
    footprint: number,
    dir: Vector3D
  );
  public constructor(
    pos?: Vector3D,
    importance?: number,
    potential?: number,
    footprint?: number,
    dir?: Vector3D
  ) {
    super();
    if (pos !== undefined && importance !== undefined && potential !== undefined && footprint !== undefined && dir !== undefined) {
      this.m_pos = new Vector3D(pos.x, pos.y, pos.z);
      this.m_dir = new Vector3D(dir.x, dir.y, dir.z);
      this.SetAll(importance, potential, footprint);
    }
  }

  public Importance(): number {
    return this.m_power.r;
  }

  public PImportance(): number {
    return this.m_irradiance.r;
  }
}

File: java/base/src/main/java/vsdk/toolkit/raycasting/photonMap/Importon.java
package vsdk.toolkit.raycasting.photonMap;

import vsdk.toolkit.common.linealAlgebra.Vector3D;

// Importon: identical to IrrPhoton, but with some extra functions
public class Importon extends IrrPhoton {
    public void
    SetAll(float imp, float pot, float foot) {
        // Abuse m_power for importance estimates.
        // -- AT LEAST 3 COLOR components needed!  Watch out with compact photon repr.
        m_power.getR() = imp;
    }

    public void
    PSetAll(float imp, float pot, float foot) {
        // Abuse m_power for importance estimates.
        // -- AT LEAST 3 COLOR components needed!  Watch out with compact photon repr.
        m_irradiance.getR() = imp;
    }

    public Importon() {
        super();
    }

    public Importon(
        Vector3D pos,
        float importance,
        float potential,
        float footprint,
        Vector3D dir)
    {
        super();
        m_pos = new Vector3D(pos.x, pos.y, pos.z);
        m_dir = new Vector3D(dir.x, dir.y, dir.z);

        SetAll(importance, potential, footprint);
    }

    public float
    Importance() {
        return (float)m_power.getR();
    }

    public float
    PImportance() {
        return (float)m_irradiance.getR();
    }
}
